forgerock-authenticator/src/main/java/org/forgerock/android/auth/biometric/BiometricAuth.java

```diff
@@ -0,0 +1,217 @@
+                && this.keyguardManager.isDeviceSecure()) {
+            initDeviceCredentialAuthentication();
+        }
+    }
+
+    private void setServicesFromActivity(@NonNull FragmentActivity activity) {
+        Context context = activity.getBaseContext();
+        this.biometricManager = BiometricManager.from(activity);
+        this.keyguardManager = (KeyguardManager) context.getSystemService(KEYGUARD_SERVICE);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            this.fingerprintManager = (FingerprintManager) context.getSystemService(Context.FINGERPRINT_SERVICE);
+        }
+    }
+
+    private BiometricPrompt initBiometricPrompt() {
+        Executor executor = ContextCompat.getMainExecutor(this.activity);
+        BiometricPrompt biometricPrompt = new BiometricPrompt(
+                this.activity, executor, new BiometricPrompt.AuthenticationCallback() {
+            @Override
+            public void onAuthenticationError(int errorCode,
+                                              @NonNull CharSequence errString) {
+                super.onAuthenticationError(errorCode, errString);
+                biometricAuthListener.onError(errorCode, errString.toString());
+            }
+
+            @Override
+            public void onAuthenticationSucceeded(
+                    @NonNull BiometricPrompt.AuthenticationResult result) {
+                super.onAuthenticationSucceeded(result);
+                biometricAuthListener.onSuccess(result);
+            }
+
+            @Override
+            public void onAuthenticationFailed() {
+                super.onAuthenticationFailed();
+            }
+        });
+
+        int authenticators = BIOMETRIC_STRONG;
+        if(this.allowDeviceCredentials) {
+            authenticators += DEVICE_CREDENTIAL;
+        }
+
+        BiometricPrompt.PromptInfo.Builder builder = new BiometricPrompt.PromptInfo.Builder()
+                .setAllowedAuthenticators(authenticators);
+        if(this.title == null) {
+            builder.setTitle("Biometric Authentication for login");
+        }
+        if(this.subtitle == null) {
+            builder.setSubtitle("Log in using your biometric credential");
+        }
+        if(authenticators < DEVICE_CREDENTIAL) {
+            builder.setNegativeButtonText("Cancel");
+        }
+        this.promptInfo = builder.build();
+
+        return biometricPrompt;
+    }
+
+    private boolean hasBiometricCapability() {
+        int canAuthenticate = this.biometricManager.canAuthenticate(BIOMETRIC_STRONG | DEVICE_CREDENTIAL);
+        return canAuthenticate == BiometricManager.BIOMETRIC_SUCCESS;
+    }
+
+    private void initBiometricAuthentication() {
+        BiometricPrompt biometricPrompt = initBiometricPrompt();
+        biometricPrompt.authenticate(this.promptInfo);
+    }
+
+    private void initDeviceCredentialAuthentication() {
+        if(this.allowDeviceCredentials) {
+            DeviceCredentialFragment.launch(this);
```

Comment:
Have a look at registerForActivityResult instead of using startActivityForResult, you may not need a headless Fragment to receive the callback from onActivityResult. May able to simplify the code.

Reply:
This was my first option, but was not able to make it work. I will give a second try, because startActivityForResult is also deprecated.